src/App/Fossa/FossaAPIV1.hs

```diff
@@ -324,11 +326,17 @@ archiveUpload signedArcURI arcFile = fossaReq $ do
   let arcURL = URI.mkURI $ signedURL signedArcURI
 
   uri <- fromMaybeText ("Invalid URL: " <> signedURL signedArcURI) arcURL
-  (url, options) <- fromMaybeText ("Invalid HTTPS URI: " <> T.pack (show uri)) (useHttpsURI uri)
-  _ <- context "Uploading project archive" $ 
-    reqCb PUT url (ReqBodyFile arcFile) lbsResponse options (pure . requestEncoder)
+  validatedURI <- fromMaybeText ("Invalid URI: " <> T.pack (show uri)) (useURI uri)
+
+  _ <- context "Uploading project archive" $ case validatedURI of
+        Left (url, options) -> uploadArchiveRequest url options
+        Right (url, options) -> uploadArchiveRequest url options
```

Comment:
unfortunately this won't work -- the type of `validatedURI` is `Either (Url 'Http, Option scheme0) (Url 'Https, Option scheme1)` notably clashing on `Url 'Http` vs `Url 'Https`, as well as `scheme0` vs `scheme1`

we have a workaround we use elsewhere via `Fossa.API.Types.useApiOpts` (which internally `unsafeCoerce` s to unify these types, because they _are_ the same, but we can't convince the type system of that). probably not worth it here, though

Reply:
Yea, I was going off of `useAPIOpts` as my example. Are you saying I should leave this as is?

Reply:
oops.

Reply:
Yeah, leave this as-is